### ISSUE TITLE
Old OSPF interfaces in MySQL database not clearing out when device is rediscovered.

### ISSUE THREAD

**Issue by netnem** [librenms/librenms#7431]
I've got an alert rule set up in LibreNMS as such:

%ospf_ports.ospfIfIpAddress ~ "10.4."   && %ospf_ports.ospfIfState = "down"

It reports back a Brocade device_id (63) with ospfIfIpAddress of 10.4.0.72 and shows down.  However, this OSPF interface has been completely removed from device_id (63).  There is no OSPF configuration on the device at all for this subnet or the port that the OSPF adjacency used to be on. 

I run a new discovery on the device (./discovery.php -h 63) and the old interface still shows up in the MYSQL database (as well as the alert notifications).  

I can manually delete the row ( DELETE FROM 'librenms','ospf_ports' WHERE 'device_id'='63' and 'ospf_port_id'='10.4.0.72' ) and then run a rediscovery on the device and the OSPF port does not repopulate.

In addition, I often test OSPF configurations on a router that is at my desk, and then deploy the router in a different location for official deployment.  The OSPF interface when the new router is deployed (off-site) will remain as showing as part of the router that was at my desk.

**Comment by laf** [comment]
It could be pretty simple fix but i'd like to get some data to see: Can you post the output of `./poller.php -h HOSTNAME -d -r -f -m ospf`

**Comment by netnem** [comment]
Sure, I'll have to pull a new example using the 144.30.251 address range:

#verified ip address config does not exist on router
device15#show run | inc 144.30.251.70
device15#

#verified in SQL database:
'15', '2737', '144.30.251.70.0', '144.30.251.70', '0', '0.0.0.160', 'broadcast', 'enabled', '1', '1', '5', '10', '40', 
'120', 'down', '0.0.0.0', '0.0.0.0', '347', NULL, 'active', 'multicast', 'false', 'none', ''

./poller.php -h 15 -d -r -f -m ospf 
https://pastebin.com/duhcX8im

**Comment by laf** [comment]
PR submitted @netnem - can you test please.

**Comment by netnem** [comment]
Confirmed working.

==Details==

I was able to remove ospf config, rerun poller.php and the alert disappeared.  I also moved the ospf config from one router to another router, and the alert moved to where it was supposed to be (previously it would stay on the old router.)  I then moved the ospf connection back to the original location and the alert moved where it should be.  Fix is performing as expected.
